scripts/cot/botania/flowers/functional/rokku_eryngium.zs

```diff
@@ -0,0 +1,121 @@
+import mods.contenttweaker.VanillaFactory;
+import mods.randomtweaker.cote.ISubTileEntity;
+import mods.randomtweaker.cote.ISubTileEntityFunctional;
+import mods.randomtweaker.cote.SubTileEntityInGame;
+import mods.randomtweaker.cote.Update;
+import mods.zenutils.DataUpdateOperation.OVERWRITE;
+import native.net.minecraft.util.EnumParticleTypes;
+import native.net.minecraft.world.WorldServer;
+
+static validCrystalNames as IItemStack[string] = {
+    'item.itemcelestialcrystal'         : <item:astralsorcery:itemcelestialcrystal>,
+    'item.itemrockcrystalsimple'        : <item:astralsorcery:itemrockcrystalsimple>, 
+} as IItemStack[string];
+
+static manaCostPerCut as int = 1000;
+
+var rokku_eryngium = VanillaFactory.createSubTileFunctional("rokku_eryngium", 0xffffff);
+rokku_eryngium.hasMini = false;
+rokku_eryngium.maxMana = 1000;
+rokku_eryngium.range = 1;
+rokku_eryngium.onUpdate = function(subtile, world, pos) {
+    if(world.isRemote()
+    || world.time%20 != 5) return;
+    if(isNull(subtile.data)
+    || isNull(subtile.data.crystalProperties)
+    || subtile.data.crystalProperties.collectiveCapability == -1){
+        pickCrystal(world, pos, subtile);
+        return;
+    } else {
+        if(subtile.getMana() < manaCostPerCut) return;
+        subtile.consumeMana(manaCostPerCut);
+        workOnCrystal(world, pos, subtile);
+        return;
+    }
+};
+rokku_eryngium.onBlockHarvested = function(world, pos, state, player) {
+    if(world.isRemote()) return;
+    val subtile = world.getSubTileEntityInGame(pos);
+    if(!isNull(subtile.data)
+    && !isNull(subtile.data.crystalProperties)
+    && subtile.data.crystalProperties.collectiveCapability != -1) dropCrystal(world, pos, subtile);
+    return;
+};
+rokku_eryngium.register();
+
+function pickCrystal(world as IWorld, pos as IBlockPos, subtile as SubTileEntityInGame) as void{
+    val crystal = findCrystal(world, pos);
+    if(isNull(crystal)) return;
+    crystal.setDead();
+    subtile.setCustomData(crystal.item.tag.astralsorcery + {name: crystal.item.name, status: "work"});
+    crystal.setDead();
+    scripts.lib.sound.play("minecraft:entity.item.pickup", pos, world);
+    return;
+}
+
+function findCrystal(world as IWorld, pos as IBlockPos) as IEntityItem{
+    val items = world.getEntityItems();
+    for item in items{
+        if(isNull(item)
+        || isNull(item.item)
+        || !item.isAlive()
+        || !(validCrystalNames has item.item.name)
+        || Math.abs(item.x - pos.x - 0.5 ) > 1
+        || Math.abs(item.y - pos.y ) > 1
+        || Math.abs(item.z - pos.z - 0.5 ) > 1
+        || item.item.tag.astralsorcery.crystalProperties.collectiveCapability==100) continue;
+        return item;
+    }
+    return null;
+}
+
+function workOnCrystal(world as IWorld, pos as IBlockPos, subtile as SubTileEntityInGame) as void{
+    /*if(subtile.data.crystalProperties.collectiveCapability==100){
+        dropCrystal(world, pos, subtile);
+        return;
+    }*/
```

Comment:
Please remove unused comments or describe how they could be useful if they supposed to be uncommented.